src/test/java/net/openhft/chronicle/network/UberHandlerTest.java

```diff
@@ -237,6 +240,92 @@ public void testBusyWritingHandlersAreCalledFirstInRoundRobin() throws IOExcepti
+                            w.writeDocument(true, d -> sendHandler(d, FIRST_CID, new RejectingOnInitializeHandler())));
+                    expectException("handler == null, check that the Csp/Cid has been sent");
+                    nc.wireOutPublisher().publish(w ->
+                            w.writeDocument(false, d -> sendMessageToHandler(d, FIRST_CID)));
+
+                    // This handler should reject in onRead
+                    nc.wireOutPublisher().publish(w ->
+                            w.writeDocument(true, d -> sendHandler(d, FIRST_CID + 1, new RejectingOnReadHandler())));
+                    nc.wireOutPublisher().publish(w ->
+                            w.writeDocument(false, d -> sendMessageToHandler(d, FIRST_CID + 1)));
+                }
+            });
+
+            long startTime = System.currentTimeMillis();
+            while (System.currentTimeMillis() - startTime < 2_000 && !Thread.currentThread().isInterrupted()) {
+                assertFalse("onWrite was called after a handler threw a RejectedHandler exception", REJECTED_HANDLER_ONWRITE_CALLED.get());
+                assertFalse("onRead was called after a handler threw a RejectedHandler exception", REJECTED_HANDLER_ONREAD_CALLED.get());
+            }
+            RUNNING.set(false);
+        }
+    }
+
+    private static class RejectingOnInitializeHandler extends AbstractSubHandler<MyClusteredNetworkContext>
+            implements WritableSubHandler<MyClusteredNetworkContext>, Marshallable {
+
+        @Override
+        public void onRead(@NotNull WireIn inWire, @NotNull WireOut outWire) {
+            REJECTED_HANDLER_ONREAD_CALLED.set(true);
+        }
+
+        @Override
+        public void onInitialize(WireOut outWire) throws RejectedExecutionException {
+            throw new RejectedExecutionException("Rejected in onInitialize");
+        }
+
+        @Override
+        public void onWrite(WireOut outWire) {
+            REJECTED_HANDLER_ONWRITE_CALLED.set(true);
+        }
+    }
+
+    private static class RejectingOnReadHandler extends AbstractSubHandler<MyClusteredNetworkContext>
+            implements WritableSubHandler<MyClusteredNetworkContext>, Marshallable {
+
+        boolean hasRejected = false;
+
+        @Override
+        public void onRead(@NotNull WireIn inWire, @NotNull WireOut outWire) {
+            if (!hasRejected) {
+                hasRejected = true;
+                throw new RejectedHandlerException("Rejected in onRead");
+            } else {
+                REJECTED_HANDLER_ONREAD_CALLED.set(true);
+            }
+        }
+
+        @Override
+        public void onInitialize(WireOut outWire) throws RejectedExecutionException {
+            // Do nothing
+        }
+
+        @Override
+        public void onWrite(WireOut outWire) {
+            if (hasRejected) {
+                REJECTED_HANDLER_ONWRITE_CALLED.set(true);
+            }
+        }
+    }
+
```

Comment:
Would have loved to have written this as a unit test instead, but the structure of these classes make that tricky